File: 백준/Gold/5430. AC/AC.py
import sys
from collections import deque

input = sys.stdin.readline

def solve():
    T = int(input())
    for _ in range(T):
        p = input().rstrip()
        n = int(input())
        arr_str = input().rstrip()

        # 1) 입력 파싱
        if n:
            arr = deque(map(int, arr_str[1:-1].split(',')))
        else:
            arr = deque()

        rev = False  # 역순 여부 플래그
        error = False

        # 2) 명령 처리
        for cmd in p:
            if cmd == 'R':
                rev = not rev
            else:  # D
                if not arr:
                    print("error")
                    error = True
                    break
                if rev:
                    arr.pop()
                else:
                    arr.popleft()

        if error:
            continue

        # 3) 최종 출력
        if rev:
            arr.reverse()
        print("[" + ",".join(map(str, arr)) + "]")

if __name__ == '__main__':
    solve()

File: 백준/Gold/5430. AC/AC.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int T = Integer.parseInt(br.readLine());
        while(T-- > 0) {
            String p = br.readLine();
            int n = Integer.parseInt(br.readLine());
            String arrStr = br.readLine();
            Deque<Integer> dq = new ArrayDeque<>();
            String[] nums = arrStr.substring(1, arrStr.length() - 1).split(",");
            for(int i = 0; i < n; i++) {
                if(!nums[i].isEmpty()) dq.add(Integer.parseInt(nums[i]));
            }
            boolean rev = false, err = false;
            for(char c : p.toCharArray()) {
                if(c == 'R') rev = !rev;
                else {
                    if(dq.isEmpty()) { err = true; break; }
                    if(rev) dq.pollLast();
                    else dq.pollFirst();
                }
            }
            if(err) sb.append("error\n");
            else {
                sb.append("[");
                while(!dq.isEmpty()) {
                    sb.append(rev ? dq.pollLast() : dq.pollFirst());
                    if(!dq.isEmpty()) sb.append(",");
                }
                sb.append("]\n");
            }
        }
        System.out.print(sb);
    }
}